### PR TITLE
docs(model-server): properly document how to start a model-server

### PR DESCRIPTION
**Pull request by nkoester**

### CODE REVIEW

File: docs/global/modules/core/pages/howto/usage-model-server.adoc [hunk 1]
```diff
@@ -1,27 +1,220 @@
...
+
+
+== Running a `model-server`
+
+The following list gives an overview of the many ways to run a `model-server`:
+
+
+=== 1. Docker
+
+We publish a docker container of the `model-server` over on https://hub.docker.com/r/modelix/model-server/tags[docker hub].
```
Comment (odzhychko) [inline docs/global/modules/core/pages/howto/usage-model-server.adoc:22]:
```suggestion
We publish a docker container of the `model-server` over on https://hub.docker.com/r/modelix/model-server/tags[Docker Hub^].
```

File: docs/global/modules/core/pages/howto/usage-model-server.adoc [hunk 2]
```diff
@@ -1,27 +1,220 @@
...
+
+=== 5. MPS
+
+You can run a *light version* of the `model-server` directly in MPS via the dedicated *MPS as Modelix Model Server Plugin*, which is published over at the https://plugins.jetbrains.com/plugin/22834-mps-as-modelix-model-server[JetBrains Marketplace].
+
+
+[IMPORTANT]
+====
+A light model-server does not provide the full features as an independent `model-server` instance will.
+More *advanced features* (e.g. xref:core:howto/modelql-writing.adoc[ModelQL], or the xref:core:howto/metrics.adoc[metrics]) and *advanced clients* (e.g. xref:core:howto/modelql.adoc[ModelQLClient]) will not work.
```
Comment (odzhychko) [inline docs/global/modules/core/pages/howto/usage-model-server.adoc:203]: Actually, the light-model-server does support ModelQL

> The `model-server` and the `mps-model-server-plugin` already implement this endpoint.
modelql.adoc

**Review by odzhychko — CHANGES_REQUESTED** [review]

**Review by odzhychko — CHANGES_REQUESTED** [review]

**Review by odzhychko — DISMISSED** [review]

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
+= How-To run the light `model-server` in MPS as plugin
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:1]:
```suggestion
= How-To run the light `model-server` in MPS as a plugin
```

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+:navtitle: mps-model-server-plugin
+
+
+You can run a *light version* of the `model-server` directly in MPS via the dedicated *MPS as Modelix Model Server Plugin*, which is published over at the https://plugins.jetbrains.com/plugin/22834-mps-as-modelix-model-server[JetBrains Marketplace^].
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:5]: Maybe we could create an explanation page that explains the difference between light and non-light model-server that can be linked here instead?

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+NOTE: More information on the component in general can be found in the xref:core:reference/component-mps-model-server-plugin.adoc[corresponding component reference].
+
+[IMPORTANT]
+====
+A light model-server does not provide the full features as an independent `model-server` instance will.
+More *advanced features* will not work with this light version (e.g. modelix specific repositories/branches, or the xref:howto/metrics.adoc[metrics]).
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:12]:
```suggestion
More *advanced features* will not work with this light version (e.g., modelix specific repositories/branches, or the xref:howto/metrics.adoc[metrics]).
```

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+If you want the full capabilities of the `model-server`, check out xref:howto/usage-model-api-gen-gradle.adoc[how to start it standalone].
+====
+
+
+== How to Install the Plugin
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:18]:
```suggestion
== How to install the Plugin
```

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+In MPS navigate to File -> Settings -> Plugins (left) -> Marketplace (top) -> Search for "modelix model server" -> Press *Install* -> Press *Restart IDE*.
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:20]: There's a dedicated syntax for such menu selections: https://docs.asciidoctor.org/asciidoc/latest/macros/ui-macros/

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+image::model-server-plugin-marketplace.png[Installing the MPS model-server plugin]
+
+You can access the light `model-server` via the
+xref:reference/component-light-model-client.adoc[light-model-client].
+
+
+== How to Run the Plugin
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:28]:
```suggestion
== How to run the plugin
```

File: docs/global/modules/core/pages/howto/usage-model-client-v2.adoc [hunk 9]
```diff
@@ -0,0 +1,105 @@
+= How-To use the `model-client` v2 (Kotlin)
+:navtitle: Use the `model-client` v2 (Kotlin)
+
+NOTE: In order to use the client, you will need a xref:core:howto/usage-model-server.adoc[running model-server] to connect to.
+
+== Gradle Dependency
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/usage-model-client-v2.adoc:6]: We should somehow decide on how to do casing. I'm always more in favor of NOT using title case, which is less recognizable to humans than normal sentences

File: docs/global/modules/core/pages/howto/usage-model-client-v2.adoc [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+
+== Usage
+
+Once set up, creating an instance that loads the entire model from the server can be done like this:
+
+
+[source, kotlin]
+--
+
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/usage-model-client-v2.adoc:33]:
```suggestion
```

File: docs/global/modules/core/pages/howto/usage-model-server.adoc [hunk 11]
```diff
@@ -1,27 +1,201 @@
...
 
-
 NOTE: If you are interested in a more practical usage of what is presented here, check out the https://github.com/modelix/modelix.samples[samples project^]
 
+NOTE: To run a light version of the `model-sever`, check out  xref:core:howto/mps-model-server-plugin.adoc[]
+
+== Backends: In Memory vs. Database
+
+The `model-server` will by default require a database backend.
+The `-jdbcconf` flag allows you to provide a custom jdbc configuration file.
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/usage-model-server.adoc:11]:
```suggestion
The `-jdbcconf` flag allows you to provide a custom JDBC configuration file.
```

File: docs/global/modules/core/pages/howto/usage-model-server.adoc
Comment (languitar) [inline docs/global/modules/core/pages/howto/usage-model-server.adoc:1]: Should we try to name the files close to the actual H1 of the content?

File: docs/global/modules/core/pages/howto/usage-model-server.adoc [hunk 12]
```diff
@@ -1,27 +1,201 @@
...
+
+
+== Running a `model-server`
+
+The following list gives an overview of the many ways to run a `model-server`:
+
+
+=== 1. Docker
+
+We publish a docker container of the `model-server` over on https://hub.docker.com/r/modelix/model-server/tags[Docker Hub^].
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/usage-model-server.adoc:24]:
```suggestion
We publish a Docker container of the `model-server` over on https://hub.docker.com/r/modelix/model-server/tags[Docker Hub^].
```

and so on

File: docs/global/modules/core/pages/howto/usage-model-server.adoc [hunk 13]
```diff
@@ -1,27 +1,201 @@
...
+To run the model-server container with the in memory backend, simply call the following.
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/usage-model-server.adoc:25]:
```suggestion
To run the model-server container with the in-memory backend, simply call the following.
```

File: docs/global/modules/core/pages/howto/usage-model-server.adoc [hunk 14]
```diff
@@ -1,27 +1,201 @@
...
+        interval: 10s
+        timeout: 3s
+        retries: 10
+      volumes:
+        - model-server-db:/var/lib/postgresql/data
+        - ./init-database.sql:/docker-entrypoint-initdb.d/init-database.sql:z
+
+volumes:
+  model-server-db: {}
+  dashboard-db: {}
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/usage-model-server.adoc:100]:
```suggestion
```

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc (hunk 4 above)
Comment (nkoester) [reply docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:5]: sounds reasonable. but beyond this PR which provides a how-to only.

File: docs/global/modules/core/pages/howto/usage-model-client-v2.adoc (hunk 9 above)
Comment (nkoester) [reply docs/global/modules/core/pages/howto/usage-model-client-v2.adoc:6]: yes, once we decide we have to rework all available docs

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc (hunk 4 above)
Comment (languitar) [reply docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:5]: Maybe we can at least lift the light-model-server information into a second how-to a la "how to use MPS as a model-server" or something like that?

Comment (nkoester) [reply docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:5]: I am confused - this file is exactly what you suggest?

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc (hunk 6 above)
Comment (nkoester) [reply docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:18]: All other docs use title case. I will not break this schema in this single page.

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc (hunk 8 above)
Comment (nkoester) [reply docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:28]: All other docs use title case. I will not break this schema in this single page.

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc (hunk 6 above)
Comment (languitar) [reply docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:18]: Then the H1 also needs to be title case. That's where I came from.

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+A light model-server does not provide the full features as an independent `model-server` instance will.
+More *advanced features* will not work with this light version (e.g., modelix specific repositories/branches, or the xref:howto/metrics.adoc[metrics]).
+
+If you want the full capabilities of the `model-server`, check out xref:howto/usage-model-api-gen-gradle.adoc[how to start it standalone].
+====
+
+
+== How to Install the Plugin
+
+In MPS navigate to "File > Settings > Plugins (left) > Marketplace (top) > Search for "modelix model server" > Install > Restart IDE".
```
Comment (languitar) [inline docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:20]:
```suggestion
In MPS navigate to menu:File[Settings > Plugins (left) > Marketplace (top)].
Search for "modelix model server" and press btn:[Install] and afterwards btn:[Restart IDE].
```

That would be what the documentation says for the syntax.

**Review by languitar — APPROVED** [review]

File: docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc (hunk 15 above)
Comment (nkoester) [reply docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc:20]: The shorthand syntax I used is also correct according to the docs

Comment (nkoester) [reply docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc]: I transitioned it to the menu macro syntax anyway

Comment (languitar) [reply docs/global/modules/core/pages/howto/mps-model-server-plugin.adoc]: > The shorthand syntax for menu is not on a standards track. You can use it for transient documents, but do not rely on it long term.

**Review by languitar — APPROVED** [review]

**Pull request merged**